### ISSUE TITLE
Open preview programatically

### ISSUE THREAD

**Issue by mzellho** [lukasz-galka/ngx-gallery#195]
I am trying to open your really nice `ngx-preview-component` from a click on a button.  In my component I set up the options, the image and the @ViewChild for the gallery:

```
@ViewChild('gallery') gallery: NgxGalleryComponent;

public galleryOptions: INgxGalleryOptions[] = [{
    image:      false,
    thumbnails: false,

    previewCloseOnClick:       true,
    previewCloseOnEsc:         true,
    previewDownload:           true,
    previewFullscreen:         true,
    previewKeyboardNavigation: true,
    previewSwipe:              true,
    previewZoom:               true
}];

public galleryImages: INgxGalleryImage[]    = [];
```

The template contains:
```
<ngx-gallery #gallery
             [options]="galleryOptions" 
             [images]="galleryImages" 
             (previewClose)="onGalleryClosed()">
</ngx-gallery>
```

And in the click handler for my button:

```
this.galleryImages = myArrayOfImageUrls.map(url => <INgxGalleryImage> {
    small: url,
    medium: url,
    big: url
});

this.gallery.openPreview(0);
```

I tried several things, including subscribing to `this.gallery.imagesReady` but all of them lead me to a
`image is undefined`.

If `image` is set to `true` and I click on the image, the preview opens as it should.

The next thing I was thinking of was to use `ngx-preview-component` standalone, but unfortunately, `ngx-preview-componet` is not an exported member from `NgxGalleryModule`.

Any other ideas?
Thanks a lot!

**Comment by lukasz-galka** [comment]
Have you seen my example?
https://lukasz-galka.github.io/ngx-gallery-demo/#only-preview

https://github.com/lukasz-galka/ngx-gallery-demo/blob/master/src/app/app.component.ts#L27
https://github.com/lukasz-galka/ngx-gallery-demo/blob/master/src/app/app.component.ts#L195
https://github.com/lukasz-galka/ngx-gallery-demo/blob/master/src/app/app.component.ts#L215

https://github.com/lukasz-galka/ngx-gallery-demo/blob/master/src/app/app.component.html#L43

**Comment by mzellho** [comment]
Hi @lukasz-galka , thanks for your reply.

In fact I somehow missed that example - sorry for that -, but after all I was more or less doing the same things you demonstrated in your `onlyPreview` and `async` examples.

Meanwhile I got it working. There must have been a race condition after assigning the (new) images to the gallery, which I worked around by opening the preview in a setTimeout Block:

```
this.galleryImages = myArrayOfImageUrls.map(url=> {
  return {
    // small:       url,
    // medium:      url,
    big: url
  };
});

setTimeout(() => {
             this.gallery.openPreview(0);
           }, 0
);
```

To me this is issue is resolved, but still: am I doing anything wrong here?

Thanks!
